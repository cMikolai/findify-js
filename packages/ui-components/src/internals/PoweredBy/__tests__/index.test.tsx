import * as React from 'react';
import { PoweredBy, Props } from '..';

describe('<PoweredBy />', () => {
  it('renders correctly', () => {
    const component = render(<PoweredBy />);
    expect(component).toMatchSnapshot();
  });

  it('attaches and calls onClick event handler', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<PoweredBy onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.simulate('click');
    expect(onClick.mock.calls.length).toBe(1);
  });
});
