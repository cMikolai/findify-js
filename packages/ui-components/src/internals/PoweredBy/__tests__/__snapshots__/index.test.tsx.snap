// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PoweredBy /> attaches and calls onClick event handler 1`] = `
<Component
  onClick={[Function]}
/>
`;

exports[`<PoweredBy /> renders correctly 1`] = `
<div
  class="root"
>
  Search powered by Findify
</div>
`;
